sys-devel/binutils/files/binutils-2.41-linker-prefix.patch

```diff
@@ -0,0 +1,56 @@
+https://bugs.gentoo.org/892549
+https://github.com/gentoo/binutils-gdb/pull/5
+
+From 531bc1536dfc3070c1f7d03e98fa14bcb81fdcd0 Mon Sep 17 00:00:00 2001
+From: James Le Cuirot <chewi@gentoo.org>
+Date: Sat, 23 Dec 2023 13:48:10 +0000
+Subject: [PATCH 2/2] ld: Allow a Gentoo-specific prefix to override the prefix
+ used by ld.bfd
+
+This prefix is only used to locate $prefix/etc/ld.so.conf, with $prefix
+usually being /usr. This file is important on Gentoo Prefix systems,
+where the /usr prefix is within another directory. The problem is that
+Gentoo already passes the same directory as the sysroot, and ld.bfd
+therefore looks for /myprefix/myprefix/usr/etc/ld.so.conf.
+
+The sysroot is dynamic, while the prefix is hardcoded. A hardcoded
+prefix that isn't just /usr is unhelpful, not just because of the
+doubled prefix issue above, but also because it prevents ld.bfd from
+working effectively outside its native environment. We will therefore
+hardcode it to just /usr.
+
+This change does not simply do that though and sets up a $gentoo_prefix
+variable instead, with $prefix as a fallback. This is necessary because
+Gentoo prefix-guest systems, which use the host's libc, do not apply a
+sysroot like RAP prefix systems do. In that case, we must preserve the
```

Comment:
why do RAP prefix systems apply sysroots?  sysroots ought to be / for those

Reply:
We've been over this enough times already...

Reply:
hmm, i don't recall the last time we have.  does a regular rap prefix exhibit the sysroot!=/ bug?  if it does, I can probably let one build in the background to inspect

Reply:
I'm not sure what you mean by "bug" there. We changed how RAP works a year ago so that the prefix is applied using sysroot. This allowed us to remove a lot of the ugly hacks, albeit in exchange for fewer smaller ones like this. It also made it far easier to use a non-prefixed toolchain to target a prefixed environment (including cross) and for a prefixed toolchain to target other prefixed environments.

Reply:
a 'hack' is not measured in LoC, but in how much an contract is violated.  in this particular case, our system is getting told that its sysroot, onto which prefix and other standard directory variables get rebased, is not /, when it is, and only the prefix is changed.

if the toolchain lacks support for setting up a target system prefix in the way we'd like, we can extend the toolchain interface to provide that.